### PR TITLE
Task list improvements

### PR DESCRIPTION
**Pull request by pezholio**
This adds the tasklist-specific helpers to the taskList file, as well as a couple of improvements to ensure that they work with both applications and assessments. I'm concerned we may need to be a bit smarter further down the line when we want to work out if a task is completed / in-progress, so this will help us out.

### CODE REVIEW

File: server/@types/shared/models/ApprovedPremises.ts [hunk 1]
```diff
@@ -2,11 +2,13 @@
 /* tslint:disable */
 /* eslint-disable */
 
+import type { LocalAuthorityArea } from './LocalAuthorityArea';
```
Comment (richpjames) [inline server/@types/shared/models/ApprovedPremises.ts:5]: I think these type updates should be in a different commit/PR

File: server/utils/taskListUtils.ts [hunk 2]
```diff
@@ -37,22 +38,25 @@ const getCompleteSectionCount = (sections: FormSections, applicationOrAssessment
   }).length
 }
 
-const taskLink = (task: Task, applicationOrAssessment: Application | Assessment, journeyType?: JourneyType): string => {
+const isAssessment = (applicationOrAssessment: Application | Assessment): boolean => {
```
Comment (richpjames) [inline server/utils/taskListUtils.ts:41]: instead of `: boolean` I think you could do `: applicationOrAssessment is Assessment` as a [type predicate](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates). 
I don't think there's a major advantage of doing it here as `id` is defined on both

**Review by richpjames — APPROVED** [review]
LGTM

File: server/@types/shared/models/ApprovedPremises.ts (hunk 1 above)
Comment (pezholio) [reply server/@types/shared/models/ApprovedPremises.ts]: Yeah, I dunno why those haven't come in a PR
